content/blog/2020-03-06-wasm/index.md

```diff
@@ -0,0 +1,229 @@
+For me, that's what WebAssembly brings.
+
+In this article, I don't expect to talk much about the language (or binary code format) 
+itself, but implications of its compilation which makes 
+things like [Google Earth][earth] on a browser possible.
+
+<img src="google-earth.png" width="700" >
+
+[earth]: https://www.google.com/earth/
+
+## What is WebAssembly?
+WebAssembly is a binary code format to transfer web applications from the 
+server to the browser. It is incorporated in modern browsers to be used in 
+tandem with existing JavaScript applications, and uses components from 
+existing JavaScript engines to interpret and execute. It has taken the world 
+wide web by storm as,
+- it is rolled out by four major browsers (Chrome, Edge, Mozilla and Safari)
+i.e., platform independent
+- it is programming model independent
+- it is hardware independent
+
+What this means is you could write a web application in C, compile it to 
+WebAssembly and use it on any browser on any hardware. This gives much more 
+performance in general than a JavaScript program, which has type 
+classification overheads and overheads at parsing and verification.
+Existing technologies also has performance implications based on which browser 
+you target (different browsers used different techniques to optimize applications on
+browsers).
+
+## Is WebAssembly just C on browsers?
+WebAssembly does brings the performance of pre-compiled languages to web browsers. 
+But it does so side-by-side with traditional JavaScript running on the same engines.
+This allows browsers to have best of both worlds, drive the Ferrari on the race track--- 
+run performant WebAssembly when it needs performance--- or drive the Prius to the 
+grocery store--- quickly getting a simple web application up and running.
+
+With WebAssembly, you could use JavaScript for fast development, but also 
+use C where you need performance. You could use C (theoretically supports other languages too) where static typing is
+useful, but JavaScript where dynamic typing is a necessity for productivity.
+And you could combine these modules, leveraging interoperability to balance productivity and performance,
+as both these styles can now be executed in the same compiler flow.
+So WebAssembly is more than just C on browsers, it's a carefully coordinated, 
+well-designed, massive engineering effort to reconcile two worlds at odds.
+
+## How does it differ from existing tools?
+So WebAssembly is portable, but doesn't JVM (Java virtual machine) already 
+offer portability? Even though the overall goal has some similarities, Java 
+cannot be run on a browser without plugins and has doesn't have the language
+flexibilty WebAssembly provides. This is due to Java's portability coming 
+from the ability to run Java programs in different platforms, and therefore not
+designed to run multiple programs in a browser. Java also doesn't solve the issues
```

Comment:
I don't quite get what you're saying here:

> not designed to run multiple programs in a browser

What do you mean by "multiple programs"? It's also strange to say that Java wasn't designed to run in a browser… that was one of its original big uses, although it has fallen out of favor now.

Reply:
Oh, thanks, I’ve misspoken. I meant to say “programs written in multiple languages"